src/launch.ts

```diff
@@ -1,360 +1,362 @@
+        };
+
+        logger.message("Created the following debug config:\n   type = " + debugConfig.type +
+                       "\n   cwd = " + debugConfig.cwd + " (= " + this.getLaunchTargetDirectory() + ")" +
+                       "\n   args = " + args.join(" ") +
+                       "\n   program = " + debugConfig.program + " (= " + this.getLaunchTargetPath() + ")" +
+                       "\n   MIMode = " + debugConfig.MIMode +
+                       "\n   miDebuggerPath = " + debugConfig.miDebuggerPath +
+                       "\n   stopAtEntry = " + debugConfig.stopAtEntry +
+                       "\n   symbolSearchPath = " + debugConfig.symbolSearchPath);
+
+        return debugConfig;
+    }
+
+    async validateLaunchConfiguration(op: make.Operations): Promise<string> {
+        // Cannot debug the project if it is currently building or (pre-)configuring.
+        if (make.blockedByOp(op)) {
+            return LaunchStatuses.blocked;
+        }
+
+        if (configuration.getBuildBeforeLaunch()) {
+            let currentBuildTarget: string = configuration.getCurrentTarget() || "";
+            logger.message(`Building current target before launch: "${currentBuildTarget}"`);
+            let buildSuccess: boolean = (await make.buildTarget(make.TriggeredBy.buildTarget, currentBuildTarget, false)) === make.ConfigureBuildReturnCodeTypes.success;
+            if (!buildSuccess) {
+                logger.message(`Building target "${currentBuildTarget}" failed.`);
+                let noButton: string = localize("No", "No");
+                let yesButton: string = localize("Yes", "Yes");
+                const message: string = localize("build.failed.continue.anyway", "Build failed. Do you want to continue anyway?");
+                const chosen: vscode.MessageItem | undefined = await vscode.window.showErrorMessage<vscode.MessageItem>(message,
+                {
+                    title: yesButton,
+                    isCloseAffordance: false,
+                },
+                {
+                    title: noButton,
+                    isCloseAffordance: true
+                });
+
+                if (chosen === undefined || chosen.title === noButton) {
+                    return LaunchStatuses.buildFailed;
+                }
+             }
+         }
+
+        let currentLaunchConfiguration: configuration.LaunchConfiguration | undefined = configuration.getCurrentLaunchConfiguration();
+        if (!currentLaunchConfiguration) {
+            // If no launch configuration is set, give the user a chance to select one now from the quick pick
+            // (unless we know it's going to be empty).
+            if (configuration.getLaunchTargets().length === 0) {
+                vscode.window.showErrorMessage(localize("cannot.op.no.launch.config.list.launch.targets.empty", "Cannot '{0}' because there is no launch configuration set and the list of launch targets is empty. " +

```

Comment:
I don't see this key in the generated files. I was worried because of the `+` between strings.  I don't know if that works or if you need to regenerate.

Reply:
I removed the "+" pattern, just in case.